base_rest_abstract_attachment/services/abstract_attachable.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2021 Akretion (https://www.akretion.com).
+# @author Pierrick Brun <pierrick.brun@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+import os.path
+from io import BytesIO
+
+import requests
+
+from odoo import _
+from odoo.exceptions import MissingError
+from odoo.http import content_disposition, request
+
+from odoo.addons.base_rest import restapi
+from odoo.addons.base_rest_pydantic.restapi import PydanticModel
+from odoo.addons.component.core import AbstractComponent
+
+from ..pydantic_models.attachment import AttachmentInfo, AttachmentRequest
+
+
+class AbstractAttachableService(AbstractComponent):
+    """Abstract service to allow to use attachments on a service."""
+
+    _name = "abstract.attachable.service"
+    _inherit = "base.rest.service"
+    _description = __doc__
+
+    @restapi.method(
+        routes=[(["/<int:object_id>/attachments/<int:attachment_id>"], "GET")],
+        output_param=restapi.BinaryData(required=True),
+    )
+    def download_attachment(self, object_id=None, attachment_id=None):
+        record = self._get(object_id)
+        self._check_attachment_access(record)
+        attachment = self._get_attachment_for_record(record, attachment_id)
+        content = None
+        headers = [("X-Content-Type-Options", "nosniff")]
+        if attachment.type == "url":
+            r = requests.get(attachment.url)
```

Comment:
```suggestion
        content, headers = self._get_attachment_response_data(attachment)
```
and move everything there